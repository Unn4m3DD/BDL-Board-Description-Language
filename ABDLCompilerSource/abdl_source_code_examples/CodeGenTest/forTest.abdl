main:
    let i = 2;
    for x from 0 to width do
        let i = 3;
        i = i + x;
    done;
    i = 10;
end main;