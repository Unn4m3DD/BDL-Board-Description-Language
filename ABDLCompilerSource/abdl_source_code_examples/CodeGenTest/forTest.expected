import ABDLVar from "./AbdlVar"
const sleep = (milliseconds) => {
  return new Promise(resolve => setTimeout(resolve, milliseconds))
}
const range = (a, b) => {
  const result = [];
  for (let i = a; i <= b; i++) result.push(i);
  return result;
}
const move = (origin, destination, table) => {
  try {
    table[origin.getValue()[0]][origin.getValue()[1]].onclick();
    render(context, table)
    table[destination.getValue()[0]][destination.getValue()[1]].onclick();
    render(context, table)
    await sleep(1000);
    return new ABDLVar(1);
  } catch (e) {
    return new ABDLVar(0);
  }
}
const can_move = (origin, destination, table) => {
  if (!table[origin.getValue()[0]][origin.getValue()[1]]) return new ABDLVar(0);
  table[origin.getValue()[0]][origin.getValue()[1]].onclick();
  render(context, table)
  if (!!table[destination.getValue()[0]][destination.getValue()[1]].onclick)
    return new ABDLVar(1);
  else
    return new ABDLVar(0);
}
export default async (context, table) => {
  let v0 = new ABDLVar(2);
  let v1 = new ABDLVar(0);
  let v2 = new ABDLVar(context.width);
  for(let v3 of range(v1,v2)) {
      let v4 = new ABDLVar(3);
      let v5 = v4.add(v3);
      v4 = v5;
  }
  let v6 = new ABDLVar(10);
  v0 = v6;
}